sklearn/model_selection/tests/test_search.py

```diff
@@ -371,19 +373,29 @@ def test_trivial_cv_results_attr():
 def test_no_refit():
     # Test that GSCV can be used for model selection alone without refitting
     clf = MockClassifier()
-    grid_search = GridSearchCV(clf, {'foo_param': [1, 2, 3]}, refit=False)
-    grid_search.fit(X, y)
-    assert_true(not hasattr(grid_search, "best_estimator_") and
-                hasattr(grid_search, "best_index_") and
-                hasattr(grid_search, "best_params_"))
-
-    # Make sure the predict/transform etc fns raise meaningfull error msg
-    for fn_name in ('predict', 'predict_proba', 'predict_log_proba',
-                    'transform', 'inverse_transform'):
-        assert_raise_message(NotFittedError,
-                             ('refit=False. %s is available only after '
-                              'refitting on the best parameters' % fn_name),
-                             getattr(grid_search, fn_name), X)
+    for scoring in (None, ('accuracy', 'precision')):
+        grid_search = GridSearchCV(clf, {'foo_param': [1, 2, 3]}, refit=False)
+        grid_search.fit(X, y)
+        assert_true(not hasattr(grid_search, "best_estimator_") and
+                    hasattr(grid_search, "best_index_") and
+                    hasattr(grid_search, "best_params_"))
+
+        # Make sure the predict/transform etc fns raise meaningfull error msg
+        for fn_name in ('predict', 'predict_proba', 'predict_log_proba',
+                        'transform', 'inverse_transform'):
+            assert_raise_message(NotFittedError,
+                                 ('refit=False. %s is available only after '
+                                  'refitting on the best parameters'
+                                  % fn_name), getattr(grid_search, fn_name), X)
+
+    # Test that an invalid refit param raises appropriate error messages
+    for refit in ("", 5, True, 'recall', 'accuracy'):
```

Comment:
please prefer lists, rather than tuples, where the semantics are "an ordered collection" rather than "a structured record"